plat/common/arm/cpu_native.c

```diff
@@ -82,3 +83,27 @@ void system_off(void)
 	smccc_psci_call(&smccc_arguments);
 }
 
+#ifdef CONFIG_HAVE_SMP
+/* Powers on a secondary cpu in an SMP system. */
+int cpu_on(struct lcpu *cpu, __uptr entry)
```

Comment:
Actually I don't think this would work. The entry function pointer we send to `cpu_on` is not the one in `cpu->s_args.entry`, but the common trampoline in `lcpu_start.S`. Also, the trampoline assumes it receives the whole cpu structure as an argument, so that it can initialize the stackp and jump to  `cpu->s_args.entry`.
(I will change the entry type to `__paddr_t`.)

Reply:
Yeah sure. But then make it: `int cpu_on(__lcpuid id, __paddr_t entry, void *arg)`. That is completely independent of `struct lcpu`. The point is that by taking the `lcpu` the function might suggest that it works on the `lcpu` abstraction, whereas it only uses data from it. But this function is from a design point even more low-level than this whole `lcpu` stuff. So I would provide only the necessary parameters instead of passing an object from a higher-level abstraction layer.

You would call it obviously as `cpu_on(lcpu->id, lcpu_start_paddr, lcpu);`.

Reply:
I put it in `cpu_native.c` in order to keep the psci calls in one place.